mining.md

```diff
@@ -113,59 +113,81 @@ In order to validate a block coming in from the network at round `N` was well mi
 
 ```go
 func VerifyBlock(blk Block) {
-    // 1. Verify Signature
-    pubk := GetPublicKey(blk.Miner)
-    if !ValidateSignature(blk.Signature, pubk, blk) {
-        Fatal("invalid block signature")
-    }
-
-    // 2. Verify Timestamp
-    // first check that it is not in the future
-    if blk.GetTime() > time.Now() {
-        Fatal("block was generated too far in the future")
-    }
-    // next check that it is appropriately delayed from its parents
-    if blk.GetTime() <= blk.minParentTime() + BLOCK_DELAY {
-        Fatal("block was generated too soon")
-    }
-
-    // 3. Verify ParentWeight
-    if blk.ParentWeight != ComputeWeight(blk.Parents) {
-        Fatal("invalid parent weight")
-    }
-
-    // 4. Verify Tickets
-    if !VerifyTickets(blk) {
-        Fatal("tickets were invalid")
-    }
-
-    // 5. Verify ElectionProof
-    randomnessLookbackTipset := RandomnessLookback(blk)
-    lookbackTicket := minTicket(randomnessLookbackTipset)
-    challenge := sha256.Sum(lookbackTicket)
-
-    if !ValidateSignature(blk.ElectionProof, pubk, challenge) {
-        Fatal("election proof was not a valid signature of the last ticket")
-    }
-
-    powerLookbackTipset := PowerLookback(blk)
-    minerPower := GetPower(powerLookbackTipset.state, blk.Miner)
-    totalPower := GetTotalPower(powerLookbackTipset.state)
-    if !IsProofAWinner(blk.ElectionProof, minerPower, totalPower) {
-        Fatal("election proof was not a winner")
-    }
-
-    // 6. Verify StateRoot
-    state := GetParentState(blk.Parents)
-    for i, msg := range blk.Messages {
-        receipt := ApplyMessage(state, msg)
-        if receipt != blk.MessageReceipts[i] {
-            Fatal("message receipt mismatch")
-        }
-    }
-    if state.Cid() != blk.StateRoot {
-        Fatal("state roots mismatch")
-    }
+	// 1. Verify Signature
+	pubk := GetPublicKey(blk.Miner)
+	if !ValidateSignature(blk.BlockSig, pubk, blk) {
+		Fatal("invalid block signature")
+	}
+
+	// 2. Verify Timestamp
+	// first check that it is not in the future
+	if blk.GetTime() > time.Now() {
+		Fatal("block was generated too far in the future")
+	}
+	// next check that it is appropriately delayed from its parents
+	if blk.GetTime() <= blk.minParentTime()+BLOCK_DELAY {
+		Fatal("block was generated too soon")
+	}
+
+	// 3. Verify ParentWeight
+	if blk.ParentWeight != ComputeWeight(blk.Parents) {
+		Fatal("invalid parent weight")
+	}
+
+	// 4. Verify Tickets
+	if !VerifyTickets(blk) {
+		Fatal("tickets were invalid")
+	}
+
+	// 5. Verify ElectionProof
+	randomnessLookbackTipset := RandomnessLookback(blk)
+	lookbackTicket := minTicket(randomnessLookbackTipset)
+	challenge := sha256.Sum(lookbackTicket)
```

Comment:
Why are we using `sha256` here, I am pretty sure we said we are going for blake2 everywhere?

Reply:
eh, I didnt add this in this PR, it was just there. We can change, i'm not opposed.

Reply:
I made a note in #290 to address this too